controllers/datadogagent/feature/admissioncontroller/feature.go

```diff
@@ -56,8 +56,20 @@ func (f *admissionControllerFeature) Configure(dda *v2alpha1.DatadogAgent) (reqC
 		if ac.ServiceName != nil && *ac.ServiceName != "" {
 			f.serviceName = *ac.ServiceName
 		}
+		// agent communication mode set by user
 		if ac.AgentCommunicationMode != nil && *ac.AgentCommunicationMode != "" {
 			f.agentCommunicationMode = *ac.AgentCommunicationMode
+		} else {
+			// agent communication mode set automatically
+			// use `socket` mode if either apm or dsd uses uds
+			apm := dda.Spec.Features.APM
+			dsd := dda.Spec.Features.Dogstatsd
+			if apm != nil && apiutils.BoolValue(apm.Enabled) && apiutils.BoolValue(apm.UnixDomainSocketConfig.Enabled) {
+				f.agentCommunicationMode = apicommon.AdmissionControllerSocketCommunicationMode
+			} else if dsd.UnixDomainSocketConfig != nil && apiutils.BoolValue(dsd.UnixDomainSocketConfig.Enabled) {
```

Comment:
could `dsd` be `nil` also?

Reply:
maybe instead of `if/else if` we could do 

`if (apm !=nil && ...) || (dsd != nil && ...)`  (but put them on separate lines for readability)

Reply:
DSD should also default to a non-nil value: https://github.com/DataDog/datadog-operator/blob/fe461fbaa4cfeb30857ca054753a50e618af74e0/apis/datadoghq/v2alpha1/datadogagent_default.go#L195-L197
I'll combine the `if` statements 👍